community/kernel/src/main/java/org/neo4j/kernel/builtinprocs/SchemaProcedure.java

```diff
@@ -220,19 +227,17 @@ private GraphResult getGraphResult( final Map<String,VirtualNodeHack> nodeMap,
         return graphResult;
     }
 
-    private static class VirtualRelationshipHack implements Relationship
+    private static class VirtualRelationshipHack extends RelationshipValue implements Relationship
     {
-
         private static AtomicLong MIN_ID = new AtomicLong( -1 );
 
-        private final long id;
         private final Node startNode;
         private final Node endNode;
```

Comment:
Perhaps this code would work a lot better if these were `VirtualNodeHack` objects?